teos/src/main.rs

```diff
@@ -142,12 +142,17 @@ async fn main() {
     };
     log::info!("tower_id: {tower_pk}");
 
+    let btc_rpc_auth = if conf.get_auth_method() == AuthMethod::UserPass {
+        Auth::UserPass(conf.btc_rpc_user, conf.btc_rpc_password)
+    } else {
+        Auth::CookieFile(config::data_dir_absolute_path(conf.btc_rpc_cookie))
+    };
+
```

Comment:
nit: This would be clearer for a new eye:
```suggestion
let btc_rpc_auth = match conf.get_auth_method() {
        AuthMethod::CookieFile => Auth::CookieFile(config::data_dir_absolute_path(conf.btc_rpc_cookie)),
        AuthMethod::UserPass => Auth::UserPass(conf.btc_rpc_user, conf.btc_rpc_password),
        _ => unreachable!("A verified conf will only have one of these two auth methods")
    };
```